src/mqttcm_connect.c

```diff
@@ -1462,6 +1518,69 @@ void mqtt_subscribe()
 	}
 }
 
+//Check if component name is already in the linked list
+bool CompAlreadyInList(char *comp)
+{
+	
+	MqttCMInfo("Inside CompAlreadyInList functtion\n");
+	comp_topic_name_t *current = g_head;
+	bool compExists = false;
+	while (current != NULL)
+	{
+		if (strcmp(current->compNameValue, comp) == 0)
+		{
+			//component is found in the list
+			MqttCMInfo("Component found in the list\n");
+			MqttCMInfo("The component name is %s and topic is %s\n", current->compNameValue, current->topicValue);
+			compExists = true;
+			return compExists;
+		}
+	}
+	MqttCMInfo("comp name is not present returning false\n");
```

Comment:
Move this to debug as it will confuse as an error case for the first time subscription